include/accessor.h

```diff
@@ -445,12 +363,12 @@ class accessor<DataT, Dims, Mode, target::host_task> : public detail::accessor_b
 
 	template <access_mode M = Mode, int D = Dims>
 	std::enable_if_t<detail::access::mode_traits::is_producer(M) && (D > 0), DataT&> operator[](id<Dims> index) const {
-		return *(get_buffer().get_pointer() + get_linear_offset(index));
+		return *(m_host_ptr + get_linear_offset(index));
```

Comment:
While this line is not wrong, it could just use a subscript (see device accessor `operator[]`).